yt_dlp/extractor/la7.py

```diff
@@ -57,3 +63,134 @@ def _real_extract(self, url):
+            'upload_date': '20210323',
+        },
+    }, {
+        # embed url
+        'url': 'https://www.la7.it/embed/podcast/371497',
+        'only_matching': True,
+    }, {
+        # date already in the title
+        'url': 'https://www.la7.it/propagandalive/podcast/lintervista-di-diego-bianchi-ad-annalisa-cuzzocrea-puntata-del-1932021-20-03-2021-371130',
+        'only_matching': True,
+    }, {
+        # title same as show_title
+        'url': 'https://www.la7.it/otto-e-mezzo/podcast/otto-e-mezzo-26-03-2021-372340',
+        'only_matching': True,
+    }]
+
+    def _extract_infos(self, webpage, video_id=None):
+        if not video_id:
+            video_id = self._search_regex(
+                r'data-nid=([\'"])(?P<vid>\d+)\1',
+                webpage, 'video_id', group='vid')
+
+        media_url = self._search_regex((
+            r'src:\s*([\'"])(?P<url>.+?mp3.+?)\1',
+            r'data-podcast=([\'"])(?P<url>.+?mp3.+?)\1'),
+            webpage, 'media_url', group='url')
+        ext = determine_ext(media_url)
+        formats = [{
+            'url': media_url,
+            'format_id': ext,
+            'ext': ext,
+        }]
+
+        title = self._html_search_regex((
+            r'<div class="title">(?P<title>.+?)</',
+            r'<title>(?P<title>[^<]+)</title>',
+            r'title:\s*([\'"])(?P<title>.+?)\1'),
+            webpage, 'title', group='title')
+
+        description = self._html_search_regex((
+            r'<div class="description">(.+?)</div>',
+            r'<div class="description-mobile">(.+?)</div>',
+            r'<div class="box-txt">([^<]+?)</div>',
+            r'<div class="field-content"><p>(.+?)</p></div>'),
+            webpage, 'description', default=None) or self._html_search_meta(
+                'description', webpage)
+
+        thumb = self._html_search_regex((
+            r'<div class="podcast-image"><img src="(.+?)"></div>',
+            r'<div class="container-embed"[^<]+url\((.+?)\);">',
+            r'<div class="field-content"><img src="(.+?)"'),
+            webpage, 'thumbnail', fatal=False, default=None)
+
+        duration = parse_duration(self._html_search_regex(
+            r'<span class="(?:durata|duration)">([\d:]+)</span>',
+            webpage, 'duration', fatal=False, default=None))
+
+        date = self._html_search_regex(
+            r'<div class="data">\s*(?:<span>)?([\d\.]+)\s*</',
+            webpage, 'date', default=None)
+
+        date_alt = self._search_regex(
+            r'(\d+[\./]\d+[\./]\d+)', title, 'date_alt', default=None)
+        ppn = self._search_regex(
+            r'ppN:\s*([\'"])(?P<ppn>.+?)\1',
+            webpage, 'ppn', group='ppn', default=None)
+        # if the date is not in the title
+        # and title is the same as the show_title
+        # add the date to the title
+        if date and not date_alt and ppn == title.lower():
+            title += ' del %s' % date
```

Comment:
why add date to title?

Reply:
becuase, as stated in the title and in the tests, one of the podcast available on the website has the episode title identical to the show title and that results in the single episode title to be indistinguishable from one episode to the other (expecially when downloading a playlist)

eg:
https://www.la7.it/otto-e-mezzo/podcast (ep title same as show title)
https://www.la7.it/dimartedi/podcast (normal)

```
λ py -m yt_dlp --get-title https://www.la7.it/otto-e-mezzo/podcast
Otto e Mezzo Sabato
Otto e Mezzo del 26.03.2021
Otto e Mezzo del 25.03.2021
Otto e Mezzo del 24.03.2021
Otto e Mezzo del 23.03.2021
Otto e Mezzo del 22.03.2021
Otto e Mezzo Sabato
Otto e Mezzo del 19.03.2021
Otto e Mezzo del 18.03.2021
Otto e Mezzo del 17.03.2021
```
```
λ py -m yt_dlp --get-title https://www.la7.it/otto-e-mezzo/podcast
Otto e Mezzo Sabato
Otto e Mezzo
Otto e Mezzo
Otto e Mezzo
Otto e Mezzo
Otto e Mezzo
Otto e Mezzo Sabato
Otto e Mezzo
Otto e Mezzo
Otto e Mezzo
```

instead a "normal" podcast titles are:

```
λ py -m yt_dlp --get-title https://www.la7.it/dimartedi/podcast
L'intervista di Giovanni Floris a Pier Luigi Bersani - Puntata del 23/3/2021
L'intervista di Giovanni Floris a Nicola Magrini (Direttore Aifa) - Puntata del 16/3/2021
Gli effetti del Covid sui giovani - Puntata del 9/3/2020
L'intervista di Giovanni Floris all'astrofisica Ersilia Vaudo - Puntata del 23/2/2021
L'avvocato Emmanuela Bertucci ci spiega come funzionano cashback, flat tax, cartelle esattoriali in scadenza e bonus ristrutturazioni
L'intervista di Giovanni Floris a Gennaro Sangiuliano - Puntata del 9/2/2021
L'intervista di Giovanni Floris a Evelina Christillin - Puntata del 2/2/2021
Giovanni Floris racconta la storia del ghetto di Roma - Puntata del 26/01/2021
L'intervista di Giovanni Floris a Giacomo Mameli - Puntata del 19/01/2021
ASCOLTA LA7 - Valerio Lundini
```